### PR TITLE
[FUSION-99] (iOS-Swift) Error handling improvements and code cleanup

### PR DESCRIPTION
**Pull request by akamimoor**
## JIRA ticket
[FUSION-99](https://janrain.atlassian.net/browse/FUSION-99)

## Changes

- All-around error handling has been added and improved for consistency, detail, and state management.
- Logging and alerts have been added and improved.
- Unit tests improvements.
- Code cleanup.

Update AppAuth-iOS SDK:
`carthage update --use-xcframeworks --platform iOS`

Confluence doc for development instructions: [Here](https://janrain.atlassian.net/l/cp/wqc2506v)

[FUSION-99]: https://janrain.atlassian.net/browse/FUSION-99?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Examples/Example-iOS_Swift-Carthage/ExampleTests/Cases/Managers/AuthenticatorTests.swift [hunk 1]
```diff
@@ -83,7 +83,7 @@ class AuthenticatorTests: XCTestCase {
             XCTAssertNotNil(discoveredConfig)
             
         } catch {
-            XCTFail("Expected to succeed while awaiting, but failed.")
+            XCTFail("Expected to sucddceed while awaiting, but failed.")
```
Comment (kreese-akamai) [inline Examples/Example-iOS_Swift-Carthage/ExampleTests/Cases/Managers/AuthenticatorTests.swift:86]: Typo. This should probably be spelled, _"succeed"_

**Review by kreese-akamai — COMMENTED** [review]
Build 🟢 
Run 🟢 
Test 🟢 _(updated)_

**Review by jbush-janrain — APPROVED** [review]
✅ Code review - variable renames make sense to me.  Great catch on spelling form kevin.
✅ Unit tests.  Took updating and re-running, but got them passing locally

**Review by kreese-akamai — APPROVED** [review]
LGTM 👍 
tests pass :green_circle:

**Review by kreese-akamai — APPROVED** [review]

**Pull request merged**